src/core/store.ts

```diff
@@ -423,13 +422,12 @@ export const createStore = (
           })
           .catch((e) => {
             if (e instanceof Promise) {
-              if (
-                isSuspensePromise(e) &&
-                isSuspensePromiseAlreadyCancelled(e)
-              ) {
+              if (isSuspensePromise(e)) {
                 // schedule another read later
                 // FIXME not 100% confident with this code
-                e.then(() => readAtomState(version, atom, true))
+                e.then(() => {
+                  readAtomState(version, atom, true)
```

Comment:
What's the difference here? Like readAtomState won't pass the returned value to the next `then`?

Reply:
Sorry for confusion. This doesn't mean anything special. Just making it clear that it returns void.